### DISCUSSION TITLE
Please do not fill the shell with useless texts (e.g. LOADING BOXSTARTER, HELLO WORLD, ...)

### DISCUSSION THREAD

**Discussion by schittli** [mwrock/boxstarter#517] [General]
## Is your feature request related to a problem? Please describe.
Perhaps you know old software that communicated every action, e.g. early Microsoft Word versions were very annoying:
You saved a file - and the Software reported in a message box: "file saved". You even had to confirm the message... 😕

BoxStarter is great because it can easily handle dozens of steps in a single function call.
But with dozens of actions performed, it is important that users know at the end what was done and where errors were. Without having to concentrate on trying to read all the texts presented. 😃 

If software displays a lot of unimportant information, then users no longer read it / cannot read it at all, and that important information is then overlooked.

#### Modern Concepts about "when does SW inform the user?" are much more useful:

1. In general, SW simply executes functions. Silently.

   In contrast: boxstarter generates messages like:
   `*** LOADING BOXSTARTER ***`
   `*** HELLO WORLD ***`
   » If a user aks the computer to start BoxStarter, then BoxStarter has not to tell that it is starting now. Those messages are not useful and not appropriate because the user does not get important / new information. 
   What happens is exactly what the user expects 😃  
2. Information is only generated in the event of an error.
   » So if BoxStarter could **not** start, then it is useful and appropriate to inform the user.
3. If the user asks an application to be verbose…
   Then it makes sense to report "initializing BoxStarter",
   but it does not make sense to report "BoxStarter initialized" because of no error was reported, then it is clear that everything is fine. 

## Describe the solution. Why is it needed?

**It would be great if BoxStarter could use a concept that just displays really _useful information_ which just informs about important steps.**

Because BoxStarter is so ingenious and performs so many actions, it is **very important** that the information generated makes sense and only displays the summary because those information is important:
- What was done
- Did we have any errors?

Therefore, a perfect solution would create those messages:
```
- Installing package 1
- Installing package 2
   Error: Software x can not be installed on Windows 7
- Installing package 3
   Warning: Package already installed. User -force to remove this message and enforce installation
- Installing package 4
```

## Additional Context.
N/A

## Related Issues
N/A

**Comment by pauby** [comment]
Converted to a discussion.

**Comment by mwallner** [comment]
Hi,
I 99% agree with what you're suggesting here.

The messages you've primarily mentioned (`*** LOADING BOXSTARTER *** / *** HELLO WORLD ***`) were introduced by myself when I converted the core logic of how Boxstarter integrates with Chocolatey. (In order to help me verify that everything works as I expect)
That said, I see no real reason to keep those in. (I like the idea of converting them to verbose-only output)  => put on my agenda to talk about in our next internal Boxstarter discussion round. 

As of your idea about the "perfect output" for a Boxstarter run - not sure if I like the output of Boxstarter to be vastly different as what a series of consecutive `choco install` commands would yield. ... I like to think of Boxstarter as an addition to Chocolatey (at least for installing packages).

:thinking: 
 
What we could always do is to implement a new feature/switch, that would allow the user to decide how the output/report should be presented.
